appscale/tools/appscale_tools.py

```diff
@@ -505,14 +505,26 @@ def run_instances(cls, options):
       AppScaleLogger.log("\nPlease wait for AppScale to prepare your machines "
                          "for use. This can take few minutes.")
 
+      head_node.public_ip = public_ips[0]
```

Comment:
Since the head node is already being popped and re-inserted from the list, why not just do it here to ensure it's first?
```
node_layout.nodes.remove(head_node)
node_layout.nodes.insert(0, head_node)
[iterate over entire node list]
```